gdnative-core/src/core_types/vector2.rs

```diff
@@ -89,6 +92,17 @@ impl Vector2Godot for Vector2 {
         Vector2::new(self.y, -self.x)
     }
 
+    #[inline]
+    fn move_towards(self, to: Vector2, delta: f32) -> Self {
+        let vd = to - self;
+        let len = vd.length();
+        if len <= delta || len < CMP_EPSILON {
+            to
+        } else {
+            self + vd / len * delta
```

Comment:
This implementation looks fair enough, but maybe it's appropriate to use `Vector2D::lerp` here? I believe it might behave better numerically then division when dealing with more extreme values.

Reply:
did You mean change `self + vd / len * delta` to `Vector2D::lerp(&self, to, delta)` if so it don't work, result it's not even near :(

Reply:
`lerp` expects an interpolation factor `t` between 0 and 1. That value should be `delta / len`.

Reply:
changed